src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceSourceConfig.java

```diff
@@ -161,47 +161,47 @@ public boolean isSoqlQuery() {
                                      SalesforceSourceConstants.PROPERTY_QUERY);
   }
 
-  @Override
   public void validate(FailureCollector collector) {
-    super.validate(collector);
-    if (!containsMacro(SalesforceSourceConstants.PROPERTY_QUERY) && !Strings.isNullOrEmpty(query)) {
-      if (!SalesforceQueryUtil.isQueryUnderLengthLimit(query) && SalesforceQueryParser.isRestrictedQuery(query)) {
-        collector.addFailure(
-          String.format(
-            "SOQL Query with restricted field types (function calls, sub-query fields) or "
-              + "GROUP BY [ROLLUP / CUBE], OFFSET clauses cannot exceed SOQL query length: '%d'. "
-              + "Unsupported SOQL query: '%s'", SalesforceConstants.SOQL_MAX_LENGTH, query), null)
-          .withConfigProperty(SalesforceSourceConstants.PROPERTY_QUERY);
-        throw collector.getOrThrowException();
-      }
-      SObjectDescriptor queryDescriptor;
-      try {
-        queryDescriptor = SalesforceQueryParser.getObjectDescriptorFromQuery(query);
-      } catch (SOQLParsingException e) {
-        collector.addFailure(String.format("Invalid SOQL query '%s' : %s", query, e.getMessage()), null)
-          .withStacktrace(e.getStackTrace())
-          .withConfigProperty(SalesforceSourceConstants.PROPERTY_QUERY);
-        throw collector.getOrThrowException();
-      }
-      if (canAttemptToEstablishConnection()) {
-        validateCompoundFields(queryDescriptor.getName(), queryDescriptor.getFieldsNames(), collector);
+    if (this.getConnection() != null) {
```

Comment:
some of these validations can be done even when connection is null. e.g: getObjectDescriptorFromQuery()

Skip validation only if connection is used for the validation and it's null.
e.g:
```
        if (connection != null && connection.canAttemptToEstablishConnection()) {
          validateCompoundFields(queryDescriptor.getName(), queryDescriptor.getFieldsNames(), collector);
        }
```

Also, prefer to avoid adding an extra level of indent when possible to make the code more readable.
e.g:
instead of -
if (connection != null) {
  // do a bunch of validations
}
do this -
if (connection == null) {
   return;
}
// do validations

Reply:
Noted. Removed this.